### ISSUE TITLE
Find and replace not working in split frame in branch develop (but fine in master)

### ISSUE THREAD

**Issue by hchapman** [syl20bnr/spacemacs#9700]
#### Description :octocat:
Find and replace (M-%) does not work in a frame which has been split vertically or horizontally on develop

#### Reproduction guide :beetle:
- Start Emacs
- Open a buffer (C-x C-f [filename])
- Split the frame (C-x 3) or (C-x 2)
- Try to find and replace (M-% a [ret] b [ret])

*Observed behaviour:* :eyes: :broken_heart:
Cursor moves to end of first find result, then throws error in minibuffer,
```
perform-replace: Args out of range: #<buffer [filename]>, 0, 0
```

*Expected behaviour:* :heart: :smile:
Found string marked, usual options to replace or skip appear

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.0.50
- Spacemacs: 0.200.9
- Spacemacs branch: develop (rev. 0e06bbaf)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
nil
```
**Not sure what happened here, but this is happening with many layers installed, and on a fresh .spacemacs file, too**
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES XWIDGETS LIBSYSTEMD LCMS2

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (args-out-of-range #<buffer [filename]> 0 0)
  buffer-substring-no-properties(0 0)
  perform-replace("is" "feh" t nil nil nil nil nil nil nil nil)
  query-replace("is" "feh" nil nil nil nil nil)
  funcall-interactively(query-replace "is" "feh" nil nil nil nil nil)
  call-interactively(query-replace nil nil)
  command-execute(query-replace)
```

**Comment by Alexander-Miller** [comment]
I used to run into this, too, though I never quite made the connection with splits.
Tried again now and it seems to work. Maybe an update to spacemacs/a plugin fixed things?

**Comment by hchapman** [comment]
Any idea what changed? It's still not working for me on most recent emacs (elisp-ppa) and spacemacs develop.

**Comment by hchapman** [comment]
Still doesn't work for me. Am I the only one?

**Comment by troyp** [comment]
No, this has been happening to me lately as well, on Emacs 26 (GTK) + Spacemacs develop (on Ubuntu). It's pretty annoying, since I usually prefer emacs-style interactive replace. I didn't realise it works on a single-window frame until I read your comment, though. That makes things easier, at least.

It seems to work on terminal emacs, though.

**Comment by braineo** [comment]
Wow, I also ran into this problem but never realized it was because of split frame. How did you find out?

**Comment by George9000** [comment]
Issue persists on the development branch. I too am using Emacs 26.

**Comment by AshKash** [comment]
Problem also happens on MacOX with Emacs-pretest-26.1-rc1-universal.dmg

**Comment by syl20bnr** [comment]
mmmh this is weird that it is only a bug on develop, the stack shows nothing that could come from spacemacs. Looks like an Emacs bug, to be sure can you reproduce it with `emacs -Q` ?

**Comment by braineo** [comment]
@syl20bnr I tested with `emacs -Q` and `prelude`, they are both fine. What interesting is something you can still do query-replace right after splitting panels once, but after that it does not work.

**Comment by braineo** [comment]
and using evil seems to be no problem at all by doing `%s/a/b/gc`

**Comment by braineo** [comment]
and seems you can do query replace in the newly splitted frame, but not the previous one. Is it because something happens during splitting?

**Comment by syl20bnr** [comment]
I debug it and found the culprit but I have no idea how to fix it, the issue is this line: https://github.com/emacs-mirror/emacs/blob/8a6521260dc650b4b713ea8bc71348cbe730f6e4/lisp/isearch.el#L3213

```
(sit-for 0)     ;make sure (window-start) is credible
```

which causes a redisplay.

if I pass `t` to prevent the redisplay with `(sit-for 0 t)` the bug disappear.

It's from a commit 18 year ago :-)
https://github.com/emacs-mirror/emacs/commit/c982ab21683e6c01e6488a5a101159b14fb58c56

I let you guys continue on this one with this info. I hope you'll find the culprit.

**Comment by jaseemabid** [comment]
`projectile-replace` with `SPC p R` exhibits the same issue. I'm on Linux with a pretty new spacemacs version b54ea8ef900e876a8715d54eed0f577fe93b38f4 and Emacs 26.1

**Comment by Miciah** [comment]
I suggested the following workaround in #10938:
```emacs-lisp
(setq frame-title-format nil)
```
It looks like the issue has been reported upstream as <https://debbugs.gnu.org/cgi/bugreport.cgi?bug=31586>.

**Comment by kommen** [comment]
I have submitted a pull request which fixes this issue for me: #11519

**Comment by d12frosted** [comment]
The PR was merged into `develop`. Please try again with the latest `develop`.
